core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -140,15 +141,19 @@ public static String genSegmentFileName(String segmentId, String UUID) {
 
   /**
    * Write segment file to the metadata folder of the table
+   *
    * @param tablePath table path
    * @param segmentId segment id
-   * @param UUID a UUID string used to construct the segment file name
+   * @param UUID      a UUID string used to construct the segment file name
    * @return segment file name
    */
   public static String writeSegmentFile(String tablePath, String segmentId, String UUID)
       throws IOException {
     String segmentPath = CarbonTablePath.getSegmentPath(tablePath, segmentId);
     CarbonFile segmentFolder = FileFactory.getCarbonFile(segmentPath);
+    boolean supportFlatFolder = Boolean.parseBoolean(CarbonProperties.getInstance()
```

Comment:
I think this property should be in CreateTable-TBLProperties instead of in CarbonProperty. Otherwise, user may change the configuration in separate data loading.

Reply:
Yes, Added table property `flat_folder` , and default value is false.